Plugins/Reveal/Reveal.cpp

```diff
@@ -46,28 +46,31 @@ Reveal::Reveal(Services::ProxyServiceList* services)
 Reveal::~Reveal()
 {
 }
-int32_t Reveal::HookStealthDetection(CNWSCreature* pObserverCreature, CNWSCreature* pHidingCreature, int32_t bClearLOS, int32_t* bSeen, int32_t* bHeard, int32_t bTargetInvisible)
+BOOL Reveal::HookStealthDetection(CNWSCreature* pObserverCreature, CNWSCreature* pHidingCreature, BOOL bClearLOS, BOOL* bSeen, BOOL* bHeard, BOOL bTargetHiding)
 {
     if (pObserverCreature->m_bPlayerCharacter && pHidingCreature->m_bPlayerCharacter && pHidingCreature->m_nStealthMode)
     {
         if (pObserverCreature->GetArea() == pHidingCreature->GetArea())
         {
-            if (*pHidingCreature->nwnxGet<int>(revealKey + "PARTY"))
+            auto partyReveal = pHidingCreature->nwnxGet<int>(revealKey + "PARTY");
+            if (partyReveal && *partyReveal)
             {
                 if (pObserverCreature->GetFaction()->GetLeader() == pHidingCreature->GetFaction()->GetLeader())
                 {
-                    if(*pHidingCreature->nwnxGet<int>(detectionKey + "PARTY"))
+                    if(pHidingCreature->nwnxGet<int>(detectionKey + "PARTY"))
                     {
                         *bSeen = true;
                     }
```

Comment:
Shouldn't this be something like...

```c
if(auto detectionVector = pHidingCreature->nwnxGet<int>(detectionKey + "PARTY"))
    *bSeen = *detectionVector;
else
    *bSeen = false;
```

nwnxGet returns a std:optional() which is true when it's set, to get the actual value you need to deref it.

Reply:
I fixed the revealkeys and forgot to fix the detectionkeys. 🤦

Reply:
Fixed. Underlying logic in how seen/heard is set by the plugin is unchanged to not break anything unforeseen.